### PR TITLE
HBASE-27846 Introduce hadoop image

### PR DESCRIPTION
**Pull request by ndimiduk**
Replaces https://github.com/apache/hbase-operator-tools/pull/119

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ main Compile Tests _ |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  codespell  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  detsecrets  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  markdownlint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  shelldocs  |   0m  0s |  |  No new issues.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  0s |  |  The patch does not generate ASF License warnings.  |
|  |   |   0m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase-kustomize/pull/6 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint markdownlint |
| uname | Linux d6e0ee6b1281 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | nobuild |
| Personality | /workdir/src/.yetus/personality.sh |
| git revision | main / e6a6d1a1b87d54cff7bda0b7c119bcab592d400d |
| modules | C: . U: . |
| versions | git=2.25.1 codespell=2.2.1 detsecrets=1.2.0 markdownlint=0.31.1 shellcheck=0.8.0 |
| Powered by | Apache Yetus 0.14.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ main Compile Tests _ |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  codespell  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  detsecrets  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  markdownlint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shelldocs  |   0m  0s |  |  No new issues.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  1s |  |  The patch does not generate ASF License warnings.  |
|  |   |   0m  6s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase-kustomize/pull/6 |
| Optional Tests | dupname asflicense codespell detsecrets shellcheck shelldocs hadolint markdownlint |
| uname | Linux 1d72b283c42e 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | nobuild |
| Personality | /workdir/src/.yetus/personality.sh |
| git revision | main / e6a6d1a1b87d54cff7bda0b7c119bcab592d400d |
| modules | C: . U: . |
| versions | git=2.25.1 codespell=2.2.1 detsecrets=1.2.0 markdownlint=0.31.1 shellcheck=0.8.0 |
| Powered by | Apache Yetus 0.14.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ main Compile Tests _ |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  codespell  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  detsecrets  |   0m  2s |  |  No new issues.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  hadolint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  markdownlint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shelldocs  |   0m  0s |  |  No new issues.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  0s |  |  The patch does not generate ASF License warnings.  |
|  |   |   0m  5s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase-kustomize/pull/6 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs markdownlint |
| uname | Linux a2aea8d27852 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | nobuild |
| Personality | /workdir/src/.yetus/personality.sh |
| git revision | main / a90ad11b7fce61a7cd296a015fdde338d583d1f2 |
| modules | C: . U: . |
| versions | git=2.25.1 codespell=2.2.1 detsecrets=1.2.0 markdownlint=0.31.1 shellcheck=0.8.0 |
| Powered by | Apache Yetus 0.14.1 https://yetus.apache.org |


This message was automatically generated.

File: dockerfiles/hadoop/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+FROM ${BASE_IMG_ALPINE}:${BASE_IMG_TAG_ALPINE} as alpine
+
+## -- Stage installed_jmx_exporter --
+# A stage that prepares the JMX Exporter installation directory. The jar must be in well-known
+# location so can refer to it in command yaml value.
+FROM alpine AS installed_jmx_exporter
+ARG JMX_PROMETHEUS_JAR_URL
+ADD --link ${JMX_PROMETHEUS_JAR_URL} /tmp/jmx_prometheus_javaagent.jar
+
+FROM ${BASE_IMG}:${BASE_TAG} as final
```
Comment (Apache9) [inline dockerfiles/hadoop/Dockerfile:35]: So there is already an image for hadoop 3?

**Review by Apache9 — COMMENTED** [review]
When will we generate the configuration files for hadoop?

File: dockerfiles/hadoop/Dockerfile (hunk 1 above)
Comment (ndimiduk) [reply dockerfiles/hadoop/Dockerfile:35]: Yes, the one I'm relying on is built out of their docker-hadoop-3 branch, https://github.com/apache/hadoop/tree/docker-hadoop-3

**Comment by ndimiduk** [comment]
> When will we generate the configuration files for hadoop?

Yes, in a later PR the config files are generated. I haven't brought that PR over from the existing one that's open against hbase-operator-tools, but the relevant commit is https://github.com/apache/hbase-operator-tools/pull/120/commits/a21fb87a37a1e4bb64bea38d28fded207830052f

**Review by petersomogyi — APPROVED** [review]

File: dockerfiles/hadoop/Dockerfile (hunk 1 above)
Comment (ndimiduk) [reply dockerfiles/hadoop/Dockerfile:35]: Note that as of now the upstream image is only build for `linux/amd64`.

**Comment by ndimiduk** [comment]
Thanks for the reviews!

**Pull request merged**